### ISSUE TITLE
Not Disposing SqlCommand or SqlDataReader on Caching.SqlServer

### ISSUE THREAD

**Issue by epignosisx** [aspnet/Extensions#940]
### Describe the bug
We are being hit at least weekly with this [issue](https://github.com/dotnet/corefx/issues/33070) filed in corefx. 

Going through the source code of the Caching.SqlServer package I notice that some classes that implement `IDisposable` are not wrapped in a `using` block.

https://github.com/aspnet/Extensions/blob/f2ef1b55d65feac6ad4e3dd8b04daa21f464631b/src/Caching/SqlServer/src/DatabaseOperations.cs#L215-L227

In this snippet both `SqlCommand` and the `SqlDataReader` implement `IDisposable` but they are not disposed, and looking at their implementation it looks like they are actually doing some clean up work:

https://github.com/dotnet/corefx/blob/a53ada190e2e0bd229e76ae4a6d45c13a986ab18/src/System.Data.SqlClient/src/System/Data/SqlClient/SqlCommand.cs#L777-L785

https://github.com/dotnet/corefx/blob/a53ada190e2e0bd229e76ae4a6d45c13a986ab18/src/System.Data.SqlClient/src/System/Data/SqlClient/SqlDataReader.cs#L773-L793

Is there a particular reason why this is not done?

**Comment by epignosisx** [comment]
@Tratcher I think you owe this area. Could you please review this issue? This is driving us crazy and this is the only think I have found that could potentially be the issue raised in corefx. Even if it’s not, seems like a problem. The associated PR is very simple, it is just wrapping the IDisposable objects in a using statement.

**Comment by Tratcher** [comment]
I don't see an obvious reason they weren't disposed.

@muratg for triage.

**Comment by muratg** [comment]
@JunTaoLuo could you take a look please (the PR is linked.)

**Comment by asfyra** [comment]
Is there any progress with this investigation?
We are about to put this library into production is this a stopper, or we should proceed?

Thanks in advance

**Comment by epignosisx** [comment]
I submitted a PR that was merged, but it will not be available until 3.0.

The truth is I don’t know if this fix will actually resolve the issues we’ve been having in production. 

Also, the fact that no one else have reported this makes it look it’s a problem on my side, rather than the framework, but at the same time we interact with other DBs and we have not run into this problem (following the recommendation of disposing of all ADO.NET objects).

I would really like to hear from you if this issue comes up. Maybe then, @tratcher will consider it for a servicing release.

I’m going to close this issue because the PR was accepted.
